### PR TITLE
Save signature version number and architecture in binary log

### PR DESCRIPTION
**Pull request by ldorau**
This PR depends on PR https://github.com/pmem/vltrace/pull/20

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/vltrace/21)

### CODE REVIEW

File: src/vltrace.h [hunk 1]
```diff
@@ -41,6 +41,15 @@
 #include <unistd.h>
 #include <stdbool.h>
 
+/* hardware architecture */
+enum arch {
+	ARCH_NONE = 0,
+	ARCH_x86_64 = 1,
+
+	ARCH_UNKNOWN
```
Comment (marcinslusarz) [inline src/vltrace.h]: = 255

File: src/print_event_cb.c [hunk 2]
```diff
@@ -550,8 +560,72 @@ print_header_bin(int argc, char *const argv[])
...
+	 *
+	 *	* Here follows actual log consisting of pairs:
+	 *
+	 *	unsigned		- size of the packet
+	 *	struct data_*		- data packet
+	 */
+
+	/* save syscall table signature */
+	char tab_signature[] = VLTRACE_TAB_SIGNATURE;
+	if (fwrite(tab_signature, sizeof(tab_signature), 1, OutputFile) != 1) {
```
Comment (marcinslusarz) [inline src/print_event_cb.c]: Please verify at compile time that sizeof(tab_signature) is really 12, like you specified above.

File: src/print_event_cb.c [hunk 3]
```diff
@@ -550,8 +560,72 @@ print_header_bin(int argc, char *const argv[])
...
+		return -1;
+	}
+
+	/* save version number */
+	unsigned major = VLTRACE_VERSION_MAJOR;
+	unsigned minor = VLTRACE_VERSION_MINOR;
+	unsigned patch = VLTRACE_VERSION_PATCH;
+	if (fwrite(&major, sizeof(major), 1, OutputFile) != 1 ||
+	    fwrite(&minor, sizeof(minor), 1, OutputFile) != 1 ||
+	    fwrite(&patch, sizeof(patch), 1, OutputFile) != 1) {
```
Comment (marcinslusarz) [inline src/print_event_cb.c]: We should use fixed size types.

File: src/print_event_cb.c [hunk 4]
```diff
@@ -550,8 +560,72 @@ print_header_bin(int argc, char *const argv[])
...
+		perror("uname");
+		return -1;
+	}
+
+	/* currently only x86_64 is supported */
+	unsigned architecture;
+	if (strncmp(buf.machine, STR_ARCH_x86_64, strlen(STR_ARCH_x86_64)) == 0)
+		architecture = ARCH_x86_64;
+	else
+		architecture = ARCH_UNKNOWN;
```
Comment (marcinslusarz) [inline src/print_event_cb.c]: Shouldn't we abort here? Such log file will not be usable.

**Review by marcinslusarz — CHANGES_REQUESTED** [review]

File: src/vltrace.h (hunk 1 above)
Comment (ldorau) [reply src/vltrace.h]: Done

File: src/print_event_cb.c (hunk 3 above)
Comment (ldorau) [reply src/print_event_cb.c]: Done

File: src/print_event_cb.c (hunk 4 above)
Comment (ldorau) [reply src/print_event_cb.c]: Done

File: src/print_event_cb.c (hunk 2 above)
Comment (ldorau) [reply src/print_event_cb.c]: Done

**Comment by ldorau** [comment]
Will be continued in PR #22

**Pull request closed without merging**